rfcs/rfc-0004-text-formatting-extensions.md

```diff
@@ -0,0 +1,371 @@
+| `font-weight` | Type used for font weight keys            | The value is an integer in the range from and including 50 to and including 950 in no particular increments. Variable font weights are supported.                                                                                                                                                                                                                                                                                             |
+
+Further value types may be published by Revolt in the future, and clients must implement them as they are published by Revolt to remain compliant with RSM.  
+Additionally, changes to those value types may be published by Revolt in the future, and clients must implement them as they are published by Revolt to remain compliant with RSM.
+
+Additionally, clients may publish or use their own value types, but for RSM compliance they may only be used in:
+
+-   a server that has explicitly opted in to using the client's value types using a server-side configuration mechanism introduced with a future RFC (if such a mechanism is introduced), and
+-   the saved notes channel
+
+## Standard keys
+
+This RFC immediately introduces the following standard keys.
+
+Further keys may be published by Revolt in the future, and clients must implement them as they are published by Revolt to remain compliant with RSM.
+Additionally, changes to those keys may be published by Revolt in the future, and clients must implement them as they are published by Revolt to remain compliant with RSM.
+
+Additionally, clients may publish or use their own keys, but for RSM compliance they may only be used in:
+
+-   a server that has explicitly opted in to using the client's keys using a server-side configuration mechanism introduced with a future RFC (if such a mechanism is introduced), and
+-   the saved notes channel
+
+### `color` and its alias `colour`
+
+**Value type**: `color`  
+**Description**: Sets the text colour to the specified colour. If `colour` is used instead of `color`, the key is still valid, but it must be ensured that only one of the two keys is used in a single STYLE BRACE.  
+**Example**:
+
+```markdown
+{color: red}(This text is red)
+{colour: linear-gradient(to right, red, orange)}(This text is a gradient from red to orange)
+```
+
+In HTML, this may yield the following:
+
+```html
+<span style="color: red;">This text is red</span>
+<span style="color: linear-gradient(to right, red, orange);"
```

Comment:
correction: assuming plain standard HTML, this is invalid syntax. 

gradient functions like `linear-gradient` in CSS create images, which means they can only be used in properties such as `background-image`. coloring text with a gradient therefore requires *three* css rules, that being: 

- `background-image`, with the desired gradient
- `background-clip` (or `-webkit-background-clip`) set to `text`, which forces the background image to use the text as a mask.
- `color` set to `transparent`, to hide the text itself. (i also see `-webkit-text-fill-color` being used in both revite and the new frontend, but i honestly cannot figure out *why.*)

Reply:
Example code is non-normative and for illustrative purposes